main.js

```diff
@@ -276,25 +283,130 @@ function displayLargeCard() {
     $catalog.classList.add('hidden')
     $details.classList.remove('hidden')
+    $cart.classList.remove('hidden')
+  }
+  if (viewName === 'cart') {
+    $catalog.classList.remove('hidden')
+    $details.classList.remove('hidden')
+    $cart.classlist.add('hidden')
   }
 }
 
-// function 8 - Define a function that takes a cart object and returns a
-// DOM tree containing the count of items in the cart.
 function cartCount(cart) {
   var $cartCount = document.createElement('div')
-  $cartCount.textContent = 'Cart (' + cart.item.length + ')'
+  $cartCount.classList.add('cart-count')
+  $cartCount.textContent = 'Cart (' + cart.items.length + ')'
   return $cartCount
 }
+
+var $cartCounter = document.querySelector('.cart-count-container')
+$cartCounter.addEventListener('click', openTheCart)
+
+function openTheCart() {
+  $catalog.innerHTML = ''
+  $details.innerHTML = ''
+  app.view = 'cart'
+  renderAppState()
+}
+
+function createCartItem(item) { // change to (cartItem)?
+  var $cartListContainer = document.createElement('div')
+  $cartListContainer.classList.add('container')
+
+  var $cartItemsContainer = document.createElement('div')
+  $cartItemsContainer.classList.add('container', 'cart-items-container')
+  $cartListContainer.appendChild($cartItemsContainer)
+
+  var $cartItemDiv = document.createElement('div')
+  $cartItemDiv.classList.add('container')
+  $cartItemsContainer.appendChild($cartItemDiv)
+
+  var $cartImageContainer = document.createElement('div')
+  $cartImageContainer.classList.add('.large-card-image-container')
+  $cartItemDiv.appendChild($cartImageContainer)
+
+  var $cartImage = document.createElement('img')
+  $cartImageContainer.appendChild($cartImage)
+  $cartImage.setAttribute('src', item.imageUrl)
+  $cartImage.classList.add('cart-image')
+
+  var $cartTextImageContainer = document.createElement('div')
+  $cartItemDiv.appendChild($cartTextImageContainer)
+  $cartTextImageContainer.classList.add('cart-text-image-container')
+
+  var $cartItemName = document.createElement('h5')
+  $cartTextImageContainer.appendChild($cartItemName)
+  $cartItemName.textContent = item.name
+
+  var $cartItemBrand = document.createElement('h5')
+  $cartTextImageContainer.appendChild($cartItemBrand)
+  $cartItemBrand.textContent = item.brand
+
+  var $cartItemPrice = document.createElement('h5')
+  $cartTextImageContainer.appendChild($cartItemPrice)
+  $cartItemPrice.textContent = item.price
+
+  return $cartListContainer
+}
+
+var cartObjects = app.cart.items
+
+function renderCartPage(cartObjects) {
+  var $cartHeader = document.createElement('h1')
+  var $cartListContainer = document.createElement('div')
+  $cartListContainer.classList.add('cartListContainer')
+  $cartHeader.textContent = 'Cart'
+  $cartHeader.classList.add('headerText')
+  $cartListContainer.appendChild($cartHeader)
+
+  for (var k = 0; k < cartObjects.length; k++) {
+    $cartListContainer.appendChild(createCartItem(app.cart.items[k]))
```

Comment:
Your `cartObjects` parameter should contain the items. No need to reach out to `app.cart.items`.

Reply:
![screen shot 2018-07-05 at 1 10 16 pm](https://user-images.githubusercontent.com/13921070/42345625-c8d62676-8054-11e8-82e8-2309bb92c8f2.png)